shell/platform/glfw/flutter_glfw.cc

```diff
@@ -507,18 +530,24 @@ FlutterDesktopWindowControllerRef FlutterDesktopCreateWindow(
   }
   GLFWClearCanvas(window);
 
+  // Create a state object attached to the window.
+  auto* state = new FlutterDesktopWindowControllerState();
+  state->window = window;
+  glfwSetWindowUserPointer(window, state);
+
+  // Create the resource context window before starting the engine, since it may
+  // call GLFWMakeResourceContextCurrent immediately.
+  state->resource_window = CreateResourceWindowForWindow(window);
```

Comment:
Can this be called something like `CreateShareWindow` or something similar? The purpose of the window was not clear at first glance.